lib/ecto/adapters/sql.ex

```diff
@@ -661,7 +661,7 @@ defmodule Ecto.Adapters.SQL do
               source: source, params: params, count: num_rows, operation: operation
 
       {:error, err} ->
-        case conn.to_constraints(err) do
+        case conn.to_constraints(err, [source: source]) do
```

Comment:
we haven't defined a `to_constraints/2` callback yet so we can't do this, this will break for other adapters. Maybe we should just define it but let's not do it right now.

For now, let's only change `lib/ecto/adapters/myxql/connection.ex`.